web/src/routes/CategoriesPage.tsx

```diff
@@ -75,6 +107,26 @@ const CategoriesPage = ({ city, pathname, cityCode, languageCode }: CityRoutePro
   }, [cityCode, languageCode, pathname, categoryId])
   const { data: parents, loading: parentsLoading, error: parentsError } = useLoadAsync(requestParents)
 
+  useEffect(() => {
+    setSimulateLoading(true)
+    const simulateTimer = setTimeout(() => setSimulateLoading(false), TIMER_FOR_LOADING_STATE)
+    return () => clearTimeout(simulateTimer)
+  }, [pathname])
+
+  const isLeafCategory = useMemo(() => {
+    if (!categories || !currentCategory) {
+      return false
+    }
+    try {
+      const mapModel = new CategoriesMapModel(categories)
+      return mapModel.getChildren(currentCategory).length === 0
+    } catch {
+      return false
+    }
+  }, [categories, currentCategory])
```

Comment:
❓ Can you please explain what we are doing here/why we need to simulate loading?
Why do we need a use memo? Also, `CategoriesMapModel` has an `isLeaf` method that can be used here.